pkg/datastore/applicationstore.go

```diff
@@ -268,6 +269,14 @@ func (s *applicationStore) List(ctx context.Context, opts ListOptions) ([]*model
 	return apps, cursor, nil
 }
 
+func (s *applicationStore) Update(ctx context.Context, app *model.Application) error {
```

Comment:
@sivchari Also, my point is if the function name is called UpdateXXX (UpdateApplication in this case), others may assume that this method updates the whole Application model, but in fact, we only update some fields and have to check which is updated and which is not. That is why instead of building a common update whole model, I created an Update function that fits the usecase (as you called in this function body: UpdateBasicInfo, UpdateConfiguration, UpdateSyncState)

Reply:
OK, I created a new API called UpdateApplication to communicate via gRPC, I think the naming is correct since it only scopes PipeCD applications, the API calls the Update function that fits the usecase. What do you think?